tests/smoke_tests/test_backward_compat.py

```diff
@@ -219,42 +230,55 @@ def test_multi_node_operations(self, generic_cloud: str):
-                    timeout=300)}
-            """,
-            f'{self.ACTIVATE_CURRENT} && result="$(sky jobs queue)"; echo "$result"; echo "$result" | grep {managed_job_name} | grep SUCCEEDED | wc -l | grep 2',
-            f'{self.ACTIVATE_CURRENT} && result="$(sky jobs queue)"; echo "$result"; echo "$result" | grep {managed_job_name} | grep \'CANCELLING\\|CANCELLED\' | wc -l | grep 1',
+            *self._switch_to_base(
+                # Cover jobs launched in the old version and ran to terminal states
+                launch_job(f'{managed_job_name}-old-0', 'echo hi; sleep 1000'),
+                launch_job(f'{managed_job_name}-old-1', 'echo hi'),
+                expect_status(f'{managed_job_name}-old-1',
+                              [sky.ManagedJobStatus.SUCCEEDED]),
+                expect_status(f'{managed_job_name}-old-0',
+                              [sky.ManagedJobStatus.RUNNING]),
+                f'sky jobs cancel -n {managed_job_name}-old-0 -y',
+                expect_status(f'{managed_job_name}-old-0', [
+                    sky.ManagedJobStatus.CANCELLED,
+                    sky.ManagedJobStatus.CANCELLING
+                ]),
+                # Cover jobs launched in the new version and still running after upgrade
+                launch_job(f'{managed_job_name}-0', 'echo hi; sleep 1000'),
+                launch_job(f'{managed_job_name}-1', 'echo hi; sleep 400'),
+                expect_status(f'{managed_job_name}-0',
+                              [sky.ManagedJobStatus.RUNNING]),
+                expect_status(f'{managed_job_name}-1',
+                              [sky.ManagedJobStatus.RUNNING]),
+            ),
+            *self._switch_to_current(
+                f'result="$(sky jobs queue)"; echo "$result"; echo "$result" | grep {managed_job_name} | grep RUNNING | wc -l | grep 2',
+                f'result="$(sky jobs logs --no-follow -n {managed_job_name}-1)"; echo "$result"; echo "$result" | grep hi',
+                launch_job(f'{managed_job_name}-2', 'echo hi; sleep 400'),
+                # Cover cancelling jobs launched in the new version
+                launch_job(f'{managed_job_name}-3', 'echo hi; sleep 1000'),
+                f'result="$(sky jobs logs --no-follow -n {managed_job_name}-2)"; echo "$result"; echo "$result" | grep hi',
+                f'sky jobs cancel -y -n {managed_job_name}-0',
+                f'sky jobs cancel -y -n {managed_job_name}-3',
+                expect_status(f'{managed_job_name}-1',
+                              [sky.ManagedJobStatus.SUCCEEDED]),
+                expect_status(f'{managed_job_name}-2',
+                              [sky.ManagedJobStatus.SUCCEEDED]),
+                f'result="$(sky jobs queue)"; echo "$result"; echo "$result" | grep {managed_job_name} | grep SUCCEEDED | wc -l | grep 3',
+                f'result="$(sky jobs queue)"; echo "$result"; echo "$result" | grep {managed_job_name} | grep \'CANCELLING\\|CANCELLED\' | wc -l | grep 3',
```

Comment:
we should use expect_status here too, right? We may need to wait for the jobs to cancel.

Reply:
Good catch!